components/script/dom/globalscope.rs

```diff
@@ -2396,6 +2396,26 @@ impl GlobalScope {
         InsecureRequestsPolicy::DoNotUpgrade
     }
 
+    /// Whether this document has ancestor navigables that are trustworthy
+    pub(crate) fn has_trustworthy_ancestor_origin(&self) -> bool {
+        if let Some(window) = self.downcast::<Window>() {
+            return window.Document().has_trustworthy_ancestor_origin();
+        }
+
+        false
```

Comment:
Drive by suggestion: Maybe you can use `is_some_and()` here.

Reply:
Thats a neat one! Thanks for the tip.